pkg/configuration/core/configtemplate_util_test.go

```diff
@@ -1,309 +0,0 @@
-			}},
-		},
-		want: []appsv1alpha1.ComponentConfigSpec{
-			tpl2,
-			tpl1,
-		},
-		wantErr: false,
-	}, {
-		name: "failed_test",
-		args: args{
-			comName:     "not exist component",
-			cComponents: cComponents,
-			dComponents: []appsv1alpha1.ClusterComponentDefinition{{
-				Name:        comType,
-				ConfigSpecs: []appsv1alpha1.ComponentConfigSpec{tpl1},
-			}},
-			aComponents: []appsv1alpha1.ClusterComponentVersion{{
-				ComponentDefRef: comType,
-				ConfigSpecs:     []appsv1alpha1.ComponentConfigSpec{tpl2},
-			}},
-		},
-		want:    nil,
-		wantErr: true,
-	}, {
-		name: "not_exist_and_not_failed",
-		args: args{
-			comName:     comName,
-			cComponents: cComponents,
-			dComponents: []appsv1alpha1.ClusterComponentDefinition{{
-				Name:        comType,
-				ConfigSpecs: []appsv1alpha1.ComponentConfigSpec{tpl1},
-			}},
-			aComponents: []appsv1alpha1.ClusterComponentVersion{{
-				ComponentDefRef: "not exist",
-				ConfigSpecs:     []appsv1alpha1.ComponentConfigSpec{tpl2},
-			},
-			}},
-		want: []appsv1alpha1.ComponentConfigSpec{
-			tpl1,
-		},
-		wantErr: false,
-	}}
-
-	for _, tt := range tests {
-		got, err := GetConfigTemplatesFromComponent(
-			tt.args.cComponents,
-			tt.args.dComponents,
-			tt.args.aComponents,
-			tt.args.comName)
-		require.Equal(t, err != nil, tt.wantErr)
-		require.EqualValues(t, got, tt.want)
-	}
-}
-
-func TestIsSupportConfigFileReconfigure(t *testing.T) {
-	mockTemplateSpec := func() appsv1alpha1.ComponentTemplateSpec {
-		return appsv1alpha1.ComponentTemplateSpec{
-			Name:        "tpl",
-			TemplateRef: "config",
-			VolumeName:  "volume",
-		}
-	}
-
-	type args struct {
-		configTemplateSpec appsv1alpha1.ComponentConfigSpec
-		configFileKey      string
-	}
-	tests := []struct {
-		name string
-		args args
-		want bool
-	}{{
-		name: "not_key_test",
-		args: args{
-			configTemplateSpec: appsv1alpha1.ComponentConfigSpec{
-				ConfigConstraintRef:   "cc",
-				ComponentTemplateSpec: mockTemplateSpec(),
-			},
-			configFileKey: "test_config",
-		},
-		want: true,
-	}, {
-		name: "not_exit_key_test",
-		args: args{
-			configTemplateSpec: appsv1alpha1.ComponentConfigSpec{
-				ConfigConstraintRef:   "cc",
-				ComponentTemplateSpec: mockTemplateSpec(),
-				Keys:                  []string{"config1", "config2"},
-			},
-			configFileKey: "test_config",
-		},
-		want: false,
-	}, {
-		name: "exit_key_test",
-		args: args{
-			configTemplateSpec: appsv1alpha1.ComponentConfigSpec{
-				ConfigConstraintRef:   "cc",
-				ComponentTemplateSpec: mockTemplateSpec(),
-				Keys:                  []string{"test_config", "config2"},
-			},
-			configFileKey: "test_config",
-		},
-		want: true,
-	}}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			if got := IsSupportConfigFileReconfigure(tt.args.configTemplateSpec, tt.args.configFileKey); got != tt.want {
-				t.Errorf("IsSupportConfigFileReconfigure() = %v, want %v", got, tt.want)
-			}
-		})
-	}
-}
```

Comment:
After removing the old UT code based on cd, many of the new UTs based on cmpd have not been implemented. Corresponding TODOs should be left in place so that they can be completed later.